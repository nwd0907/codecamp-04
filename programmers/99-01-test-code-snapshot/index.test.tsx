import renderer from "react-test-renderer";
import TestCodePage from ".";

test("페이지 테스트", () => {
    const component = renderer.create(<TestCodePage />);
    let tree: any = component.toJSON();
    expect(tree).toMatchSnapshot();

    renderer.act(() => {
        tree.props.onMouseEnter();
    });
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    renderer.act(() => {
        tree.props.onMouseLeave();
    });
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
