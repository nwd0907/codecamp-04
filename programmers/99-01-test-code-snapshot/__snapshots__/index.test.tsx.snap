// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`페이지 테스트 1`] = `
<a
  className="HOVERED"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  클릭하세요!
</a>
`;

exports[`페이지 테스트 2`] = `
<a
  className="HOVERED"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  클릭하세요!
</a>
`;

exports[`페이지 테스트 3`] = `
<a
  className="NORMAL"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  클릭하세요!
</a>
`;
